File: 2017_May23/performance/primes/gen_prime.py
def is_prime(x):
    s = int(x ** 0.5) + 1
    for i in xrange(2, s): 
        if x % i == 0: return False
    return True

def gen_prime(n):
    i = 2
    while n:
        if is_prime(i):
            print i
            n -= 1
        i += 1

from time import time
num_primes = 100000
print "Generating %d prime numbers..." % num_primes

start = time()
gen_prime(num_primes)
end = time()
print "Duration: ", end - start, "seconds"

File: 2017_May23/performance/primes/gen_prime.rb
def prime?(v)
    s = (v ** 0.5).to_i + 1
    2.upto(s) do |i|
        return FALSE if ((v%i) == 0)
    end
    return TRUE
end

def gen_prime(n)
    i = 2
    while n > 0
        if prime? i
            puts i
            n -= 1
        end
        i += 1
    end
end

num_primes = 100000
puts "Generating #{num_primes} prime numbers"

start = Time.now.to_f
gen_prime(num_primes)
duration = Time.now.to_f - start

puts "Duration: #{duration} seconds."
